ports/zephyr/modzephyr.c

```diff
@@ -79,6 +92,238 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_0(mod_shell_net_iface_obj, mod_shell_net_iface);
 
 #endif // CONFIG_NET_SHELL
 
+#ifdef CONFIG_DISK_ACCESS
+STATIC const mp_obj_type_t zephyr_disk_type;
+
+typedef struct _zephyr_disk_obj_t {
+    mp_obj_base_t base;
+    const char *pdrv;
+    int block_size;
+} zephyr_disk_obj_t;
+
+STATIC void zephyr_disk_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    zephyr_disk_obj_t *self = self_in;
+    mp_printf(print, "%s", self->pdrv);
+}
+
+STATIC mp_obj_t zephyr_disk_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
+    mp_arg_check_num(n_args, n_kw, 1, 1, false);
+    zephyr_disk_obj_t *self = m_new_obj(zephyr_disk_obj_t);
+    self->base.type = type;
+    self->pdrv = mp_obj_str_get_str(args[0]);
+
+    if (disk_access_init(self->pdrv) != 0) {
+        mp_raise_ValueError("disk not found");
+    }
+
+    if (disk_access_ioctl(self->pdrv, DISK_IOCTL_GET_SECTOR_SIZE, &self->block_size)) {
+        mp_raise_ValueError("unable to get sector size");
+    }
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+STATIC mp_obj_t zephyr_disk_readblocks(mp_obj_t self_in, mp_obj_t block_num, mp_obj_t buf) {
+    zephyr_disk_obj_t *self = self_in;
+    mp_buffer_info_t bufinfo;
+    int ret;
+
+    mp_get_buffer_raise(buf, &bufinfo, MP_BUFFER_WRITE);
+    ret = disk_access_read(self->pdrv, bufinfo.buf, mp_obj_get_int(block_num), bufinfo.len / self->block_size);
+    return mp_obj_new_bool(ret == 0);
```

Comment:
This function can return an int, so just returning `mp_obj_new_small_int(ret);` would be simpler and allow more information to pass to the upper layers in case of error.